src/db_api_util.ts

```diff
@@ -86,6 +86,184 @@ export type DatabaseCommandEntry = {
   failureReason?: string | null;
 };
 
+export type DatabaseThreadEntry = {
+  slackParentMessageTs: string | null;
+  channelId: string | null;
+  userId: string | null;
+  userPhoneNumber: string | null;
+  needsAttention: boolean | null;
+};
+
+export type ThreadInfo = {
+  slackParentMessageTs: string;
+  channelId: string;
+  userId: string | null;
+  userPhoneNumber: string | null;
+  age: number | null;
+};
+
+export async function newThreadToDb(
+  databaseThreadEntry: DatabaseThreadEntry
+): Promise<void> {
+  const client = await pool.connect();
+  try {
+    await client.query(
+      'INSERT INTO threads (slack_parent_message_ts, channel_id, user_id, user_phone_number, needs_attention, updated_at) VALUES ($1, $2, $3, $4, $5, NOW())',
+      [
+        databaseThreadEntry.slackParentMessageTs,
+        databaseThreadEntry.channelId,
+        databaseThreadEntry.userId,
+        databaseThreadEntry.userPhoneNumber,
+        databaseThreadEntry.needsAttention,
+      ]
+    );
+    logger.info('DBAPIUTIL.newThreadToDb: Successfully created thread');
+  } catch (error) {
+    logger.info('Failed to update threads; ignoring for now!');
+  } finally {
+    client.release();
+  }
+}
+
+export async function setThreadNeedsAttentionToDb(
+  slackParentMessageTs: string,
+  needsAttention: boolean
+): Promise<void> {
+  const client = await pool.connect();
+  try {
+    await client.query(
+      'UPDATE threads SET needs_attention = $1 WHERE slack_parent_message_ts = $2;',
```

Comment:
slack_parent_message_ts is only guaranteed to be unique within a channel, so we need to add channel to the WHERE clause here